### PR TITLE
nixos/prometheus2: replace alertmanagerURL with new alertmanagers option

### PR DESCRIPTION
**Pull request by WilliButz**
###### Motivation for this change
Prometheus2 does no longer support the command-line flag to specify
an alertmanager. Instead it now supports both service discovery and
configuration of alertmanagers in the alerting config section.

Simply mapping the previous option to an entry in the new alertmanagers
section is not enough to allow for complete configurations of an
alertmanager.

For example in a setup where an alertmanager runs under the same subdomain as prometheus but under a seperate path, this needs to be configured in the alerting section of the prometheus config and currently the module does not provide this functionality.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @basvandijk @andir @fpletz

### CODE REVIEW

**Comment by WilliButz** [comment]
@GrahamcOfBorg test prometheus2

**Review by philandstuff — APPROVED** [review]

File: nixos/modules/services/monitoring/prometheus/default.nix
Comment (Ma27) [inline nixos/modules/services/monitoring/prometheus/default.nix]: (nitpick) Isn't it sufficient to write the expression directly without `literalExample`? (Definetely not a blocker for this, just wondering as I did this in the past and it worked perfectly fine :) )

**Review by Ma27 — APPROVED** [review]
As `prometheus2` doesn't seem to support the old option, I guess the removal and change here is perfectly fine.

However I'd wait for another reviewer to merge as my stuff is mostly on NixOS 19.03 where I (admittedly) still use prometheus v1, so I don't have an environment where I can quickly test this :sweat_smile:

File: nixos/modules/services/monitoring/prometheus/default.nix
Comment (WilliButz) [reply nixos/modules/services/monitoring/prometheus/default.nix]: * Currently the option looks like this in the rendered manual:
literal-example

* leaving away `literalExample` but keeping the single quotes would result in this wrong example:
string-only

* And also leaving away the single quotes results in this:
plain

The last one made me realize, that I forgot a semicolon in the example, which I'll fix in a moment :)
I'd still go with the current version.

Comment (Ma27) [reply nixos/modules/services/monitoring/prometheus/default.nix]: I see... I was pretty sure that `example = { /* ... */ }` also gets pretty-printing, but obviously I was wrong :sweat_smile:

**Review by fpletz — APPROVED** [review]

**Pull request merged**
